lib/govuk_publishing_components/minitest/component_guide_test.rb

```diff
@@ -0,0 +1,23 @@
+module GovukPublishingComponents
+  module Minitest
+    module ComponentGuideTest
+      extend ActiveSupport::Concern
+
+      included do
+        test "renders all component guide preview pages without erroring" do
+          visit '/component-guide'
+
+          # Confirm accessibility JS test is available
+          assert_equal 'function', evaluate_script('typeof window.GOVUK.AccessibilityTest'), "AccessibilityTest javascript isn’t available"
+          assert_equal 'string', evaluate_script('typeof window.axe.version'), "aXe accessibility test library isn’t available"
+
+          all(:css, '.component-list a').map { |el| "#{el[:href]}/preview" }.each do |component|
```

Comment:
Since we're now within the Gem, can we do this smarter so we don't have to load the route page and speed up the tests?

Reply:
Considered that to be out of scope. I've written this as a drop-in replacement for what's in government-frontend.

Reply:
Fair 👍